Vectors

unknown07724 opened the pull request:
Vectors is a extension for [Vectors](https://en.wikipedia.org/wiki/Vector_%28mathematics_and_physics%29)
Blocks: Create vector with ID,direction and magnitude: creates a vector, vector direction, returns direction of the vector based on ID, vector magnitude, returns magnitude of the vector based on ID, Change Magnitude and direction, changes magnitude and direction of the vector based on ID, and List of IDs, lists all elements in IDs

unknown07724 committed:
Create timeutills.js

timeutills V1

unknown07724 committed:
Rename extensions/timeutills.js to extensions/unknown/timeutills.js

unknown07724 committed:
Update timeutills.js

unknown07724 committed:
Update timeutills.js

unknown07724 committed:
Update timeutills.js

unknown07724 committed:
Update timeutills.js

unknown07724 committed:
Create mailutils.js

unknown07724 committed:
Update mailutils.js

unknown07724 committed:
Update mailutils.js

unknown07724 committed:
Merge branch 'TurboWarp:master' into master

unknown07724 committed:
Delete extensions/unknown/mailutils.js

unknown07724 committed:
Delete extensions/unknown/timeutills.js

unknown07724 committed:
Merge branch 'TurboWarp:master' into api-toolbox

unknown07724 committed:
Create api-toolbox.js

unknown07724 committed:
Rename api-toolbox.js to api-toolbox.js

unknown07724 committed:
Update extensions.json

unknown07724 committed:
Update api-toolbox.js

unknown07724 committed:
Update api-toolbox.js

unknown07724 committed:
Update api-toolbox.js

unknown07724 committed:
Create vectors.js

unknown07724 committed:
Delete extensions/unknown/api-toolbox.js

unknown07724 committed:
Update extensions.json

unknown07724 committed:
Update vectors.js

fath11 commented:
there is no block for deleting vectors?

fath11 commented:
also please run !format

unknown07724 commented:
!format

unknown07724 commented:
> there is no block for deleting vectors?

Yes, just stop using a vector

DangoCat committed:
[Automated] Format code

unknown07724 committed:
Update vectors.js

unknown07724 commented:
!format

DangoCat committed:
[Automated] Format code

unknown07724 committed:
Update vectors.js

unknown07724 commented:
!format

DangoCat committed:
[Automated] Format code

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Merge branch 'TurboWarp:master' into vectors

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Update vectors.js

unknown07724 commented:
!format

DangoCat committed:
[Automated] Format code

unknown07724 commented:
@fath11 there is now

yuri-kiss commented on extensions/unknown/vectors.js:
```diff
@@ -0,0 +1,169 @@
+(function (Scratch) {
```
you're missing the header data

unknown07724 replied:
shit, Imma re-add that Ig

yuri-kiss commented on extensions/unknown/vectors.js:
```diff
@@ -0,0 +1,169 @@
+            opcode: "vectorCreate",
+            blockType: Scratch.BlockType.COMMAND,
+            text: Scratch.translate(
+              "create vector with ID [ID], direction [DIRECTION] and magnitude [MAGNITUDE]"
```
you have inconsistent capitalization across the blocks, the blocks should be written starting with lowercase. they are NOT sentences.

yuri-kiss commented on extensions/unknown/vectors.js:
```diff
@@ -0,0 +1,169 @@
+    }
+
+    vectorCreate(args) {
+      const id = args.ID;
```
you don't cast anything, this is poor practice

unknown07724 replied:
aren't they numbers, you can not put strings in there, and if you could, who would?

yuri-kiss commented on extensions/unknown/vectors.js:
```diff
@@ -0,0 +1,169 @@
+  }
+
+  // Vectors storage
+  const ids = [];
```
why can't this be a set

yuri-kiss commented on extensions/unknown/vectors.js:
```diff
@@ -0,0 +1,169 @@
+
+  // Vectors storage
+  const ids = [];
+  const directions = {};
```
these should be Object.create(null); or a Map

yuri-kiss commented on extensions/unknown/vectors.js:
```diff
@@ -0,0 +1,169 @@
+
+    vectorMag(args) {
+      const id = args.ID;
+      return magnitudes[id] ?? Scratch.translate("vector not found");
```
you should return an empty string or 0 here

unknown07724 replied:
do I want people to wonder why their shit isn't working? if you said yes then you are wrong, the answer is no.

Brackets-Coder replied:
> do I want people to wonder why their shit isn't working? if you said yes then you are wrong, the answer is no.

@GarboMuffin said to chill the language

yuri-kiss commented on extensions/unknown/vectors.js:
```diff
@@ -0,0 +1,169 @@
+
+    vectorDir(args) {
+      const id = args.ID;
+      return directions[id] ?? Scratch.translate("vector not found");
```
same here

yuri-kiss reviewed (changes requested):
this seems to literally just be a weird way of storing 2 dictionary's, what is the point of this extension, you can't even do any operations?

unknown07724 commented:
Maps are bullshit, it does what it needs to do very well

unknown07724 committed:
Update vectors.js

unknown07724 commented:
!format

DangoCat committed:
[Automated] Format code

GarboMuffin commented:
it would be wonderful if you would be more reasonable with your language on this issue tracker

unknown07724 commented:
it already does what it needs to do, I don't want to spend multiple hours rewriting the entire extension

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Merge branch 'TurboWarp:master' into vectors

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Update vectors.js

unknown07724 committed:
Update vectors.js

unknown07724 commented:
!format

DangoCat committed:
[Automated] Format code

unknown07724 committed:
Update vectors.js

CST1229 commented on extensions/unknown/vectors.js:
```diff
@@ -0,0 +1,180 @@
+      const id = Scratch.Cast.toString(args.ID);
+      return magnitudes[id] !== undefined
+        ? magnitudes[id]
+        : Scratch.translate("null");
```
translating block return valeus is usually not a good idea; people may compare the result of this block to the string `null` (or whatever it is in languages) and then have it break in other languages that have a different string for it.
(`null` specifically is a fairly language-neutral word so it should be fine untranslated. or follow the Scratch precedent and return an empty string; this is what lists also do for example so I don't think it would be confusing)

unknown07724 replied:
fixed

unknown07724 committed:
Merge branch 'TurboWarp:master' into vectors

unknown07724 committed:
Return empty string for undefined vector magnitudes

Change the return value for undefined magnitudes from a translation to an empty string.

unknown07724 committed:
Update vectors.js

unknown07724 commented:
!format

DangoCat committed:
[Automated] Format code

unknown07724 committed:
Update vectors.js

unknown07724 commented:
!format

DangoCat committed:
[Automated] Format code

unknown07724 commented:
> this seems to literally just be a weird way of storing 2 dictionary's, what is the point of this extension, you can't even do any operations?

you can now
